backend/repository/gsheet_repository.py

```diff
@@ -55,7 +57,7 @@ def store_dataframe(self, df: pd.DataFrame, storage_location: str, allow_create:
 
         df = df.replace([np.inf, -np.inf], np.nan)
 
-        set_with_dataframe(worksheet, df)
+        set_with_dataframe(worksheet, df, include_index=store_index, resize=True)
```

Comment:
@talkgarima by adding the parameter resize=True here.
I switched to a python package to assist saving pandas dataframes to gsheet thats where the set_with_dataframe() comes from.